auth/postgres/pat.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package postgres
+
+import (
+	"time"
+
+	"github.com/absmach/supermq/auth"
+	repoerr "github.com/absmach/supermq/pkg/errors/repository"
+)
+
+type dbPat struct {
+	ID          string    `db:"id,omitempty"`
+	User        string    `db:"user_id,omitempty"`
+	Name        string    `db:"name,omitempty"`
+	Description string    `db:"description,omitempty"`
+	Secret      string    `db:"secret,omitempty"`
+	IssuedAt    time.Time `db:"issued_at,omitempty"`
+	ExpiresAt   time.Time `db:"expires_at,omitempty"`
+	UpdatedAt   time.Time `db:"updated_at,omitempty"`
+	LastUsedAt  time.Time `db:"last_used_at,omitempty"`
+	Revoked     bool      `db:"revoked,omitempty"`
+	RevokedAt   time.Time `db:"revoked_at,omitempty"`
+}
```

Comment:
I'm getting error `view entity failed : sql: Scan error on column index 10, name "revoked_at": unsupported Scan, storing driver.Value type <nil> into type *time.Time`

Please use sql.NullTime


```suggestion
	ID          string       `db:"id,omitempty"`
	User        string       `db:"user_id,omitempty"`
	Name        string       `db:"name,omitempty"`
	Description string       `db:"description,omitempty"`
	Secret      string       `db:"secret,omitempty"`
	IssuedAt    time.Time    `db:"issued_at,omitempty"`
	ExpiresAt   time.Time    `db:"expires_at,omitempty"`
	UpdatedAt   sql.NullTime `db:"updated_at,omitempty"`
	LastUsedAt  sql.NullTime `db:"last_used_at,omitempty"`
	Revoked     bool         `db:"revoked,omitempty"`
	RevokedAt   sql.NullTime `db:"revoked_at,omitempty"`
```